### ISSUE TITLE
Error (making smart connections) with OpenAi Embeddings.

### ISSUE THREAD

**Issue by bbecausereasonss** [brianpetro/obsidian-smart-connections#756]
Finally, managed to create OpenAi (Embeddings)... I think... using Smart-Connect. However, each time I open Obsidian it crashes/freezes on (making OpenAi embeddings).

![image](https://github.com/user-attachments/assets/0897919a-5593-4133-816c-bcfda4653f56)

**Comment by brianpetro** [comment]
@bbecausereasonss it might be worth trying only one of Smart Connect app or the Smart Connections Obsidian plugin at a time until the embeddings issue gets sorted out.

While they both are able to make the embeddings, there isn't the same "offloading" integration as there was before, which could cause interference if both softwares are attempting to embed at the same time.

The "offloading" integration (improving Obsidian performance by offloading heavy processes to Smart Connect app) will be reintroduced shortly, but for now, since you're having issues with the embedding, it's probably worth trying just one of those softwares at a time. Lastly, when quitting the Smart Connect app, make sure to do it from the task tray (typically bottom right, near volume setting on windows) to ensure it is completely turned off 🌴

**Comment by bbecausereasonss** [comment]
> @bbecausereasonss it might be worth trying only one of Smart Connect app or the Smart Connections Obsidian plugin at a time until the embeddings issue gets sorted out.
> 
> While they both are able to make the embeddings, there isn't the same "offloading" integration as there was before, which could cause interference if both softwares are attempting to embed at the same time.
> 
> The "offloading" integration (improving Obsidian performance by offloading heavy processes to Smart Connect app) will be reintroduced shortly, but for now, since you're having issues with the embedding, it's probably worth trying just one of those softwares at a time. Lastly, when quitting the Smart Connect app, make sure to do it from the task tray (typically bottom right, near volume setting on windows) to ensure it is completely turned off 🌴

I have tried both ways, and it appears to be the same issue each time now. Ever since the naming changed in the embeddings (and whatever else was updated) I cannot make/open embeddings anymore, it seems to make them fine but will refuse to read them.

Here's a video of the process from starting embeddings to trying to open/read them (normal use-case) in Smart Connections.

https://www.dropbox.com/scl/fi/0d3teuzs0rsbb2uzeev3q/ErrorsReading.mp4?rlkey=p8vy3dgc1pwlo14e8jh3s5rrd&dl=0

**Comment by bbecausereasonss** [comment]
The error is ...

![image](https://github.com/user-attachments/assets/e2339bf6-9fc5-4df0-b119-74d65a11fa5b)

I have verified the Embeddings are now getting created (in the folder), but will not read...

![image](https://github.com/user-attachments/assets/9ff0d3f5-546e-490d-9537-80e55e27cda5)

**Comment by bbecausereasonss** [comment]
Not sure if this is related, but the plugin will not allow me to access (model settings) now. It's all blank, and nothing is populating. Not sure if the behavior is if embeddings can't load/aren't read; then you can't change model options? Anyway, just thought I'd mention that as well.

**Comment by bbecausereasonss** [comment]
Seems to be blocks related...

**Comment by algieg** [comment]
In the end I followed a recommendation from some other issue and reverted to [2.1.95](https://github.com/brianpetro/obsidian-smart-connections/releases/tag/2.1.95) and everything works fine. I'll try updating at some later point in time to check if it's been fixed.

**Comment by bbecausereasonss** [comment]
> In the end I followed a recommendation from some other issue and reverted to [2.1.95](https://github.com/brianpetro/obsidian-smart-connections/releases/tag/2.1.95) and everything works fine. I'll try updating at some later point in time to check if it's been fixed.

OMG you're right that fixed the issues and its working. So clearly what ever changes were made after this version broke reading/creating embeddings. Going to stick with this one for now... Thank you!

**Comment by brianpetro** [comment]
@bbecausereasonss thanks for the video. That will help me better try to reproduce the issue. 

And I'm happy to see you've got things working again in the meantime 😊🌴

**Comment by bbecausereasonss** [comment]
Wait nope. Spoke too soon. After trying to make ambeddings on a new note I created and restarting, it now won't load any of the embeddings (but does show the number).

![image](https://github.com/user-attachments/assets/36f2f73b-450b-42cd-835a-c6107cbeba56)

**Comment by brianpetro** [comment]
@bbecausereasonss have you tried deleting the entire `.smart-env` folder with the embeddings?

The only times I've encountered the "blocks" error you keep seeing is with some corrupted data in the embedding files. And it's possible the force refresh didn't clear out all the data since it doesn't actually delete the files 🌴

**Comment by bbecausereasonss** [comment]
> @bbecausereasonss have you tried deleting the entire `.smart-env` folder with the embeddings?
> 
> The only times I've encountered the "blocks" error you keep seeing is with some corrupted data in the embedding files. And it's possible the force refresh didn't clear out all the data since it doesn't actually delete the files 🌴

Yep, must have done it like 10x already between today and yesterday :)

**Comment by bbecausereasonss** [comment]
Honestly, I really don't think the app should automatically create embeddings, I think it needs to be a manual process; because I feel like when you upgrade/install and re-install it just creates errors by defaulting to the local model etc. Could be wrong.

**Comment by bbecausereasonss** [comment]
![image](https://github.com/user-attachments/assets/53993418-9315-4a3d-8604-7dcb65adb62a)

**Comment by bbecausereasonss** [comment]
What seems strange is that all of the errors,issues are quite random. I'm following the same steps but getting different issues at different points seemingly at random. One moment it works, the next it doesn't.

**Comment by brianpetro** [comment]
That last set of errors seems to be from an old version (no more `ApiAdapter` in latest version).

And I agree that a manual start should be added if the vault contains greater than a threshold of notes. But for most people with <1000 notes I think it tends to best be done automatically.

I've been trying all different combinations of settings and ways to try to trigger the issues, on both mac and windows, and I just can't seem to reproduce.

Is it possible that there is some external syncing process that might be interfering with the files? 

🌴

**Comment by bbecausereasonss** [comment]
> That last set of errors seems to be from an old version (no more `ApiAdapter` in latest version).
> 
> And I agree that a manual start should be added if the vault contains greater than a threshold of notes. But for most people with <1000 notes I think it tends to best be done automatically.
> 
> I've been trying all different combinations of settings and ways to try to trigger the issues, on both mac and windows, and I just can't seem to reproduce.
> 
> Is it possible that there is some external syncing process that might be interfering with the files?
> 
> 🌴

My main vault has always been in Dropbox. I also use Syncthings to Sync it to a different folder on my OSX/MBP, so that there aren't issues like I was experiencing before. I have just tried disabling Dropbox to see if it was causing issues.

**Comment by bbecausereasonss** [comment]
This is with the version rolled back to 2.1.9.5
Dropbox turned off
Embeddings folder nuked.
It tried to Create-embeddings
I am trying to use Small OpenAi for both notes/blocks

Then threw these error during creating blocks (I presume):

![image](https://github.com/user-attachments/assets/46ec6681-0a83-43ea-9b04-112f77c1eec9)

![image](https://github.com/user-attachments/assets/044709b6-32d8-4a07-b802-df44f17211db)

**Comment by brianpetro** [comment]
Since it seems to be a problem with the roll backed version too, I would focus on using the latest branch.

Another thing that has worked in the past with similarly difficult to debug issues is creating a small vault where you are able to reproduce the issue reliably. Then you can send me a link to that vault and I can test it out 🌴

**Comment by bbecausereasonss** [comment]
Hmm the odd thing is that the contents of my vault have not changed before/after the issues arose, but yes good idea on trying with a smaller vault.

**Comment by bbecausereasonss** [comment]
Okay so, after restarting the latest time even after above errors the vault embeddings/blocks have loaded and is letting me run inference on 2.1.9.5

**Comment by brianpetro** [comment]
@bbecausereasonss in `v2.2.66` I updated the logic for how the blocks getter (what was erroring in many of the past screenshots) works.

While it doesn't address whatever the root issue is, preventing a failure at that point may improve the overall results 🌴

**Comment by bbecausereasonss** [comment]
> @bbecausereasonss in `v2.2.66` I updated the logic for how the blocks getter (what was erroring in many of the past screenshots) works.
> 
> While it doesn't address whatever the root issue is, preventing a failure at that point may improve the overall results 🌴

I'm scared to try it again because it was just freezing completely last 2 days. Maybe I'll try again again later :)

**Comment by brianpetro** [comment]
Ok, or try with a small test vault 🌴

**Comment by bbecausereasonss** [comment]
Turning on Obsidian, next day. Nothing changed. Same issue. Until I nuked 300 notes or so from a specific Encyclopedia folder (now its loading).

Is there a embedding/vault limit of sorts or maybe a point at which it's too big?
